src/embedded_flash/syscall.rs

```diff
@@ -212,6 +212,9 @@ pub struct SyscallUpgradeStorage {
 impl SyscallUpgradeStorage {
     /// Provides access to the other upgrade partition and metadata if available.
     ///
+    /// The implementation assumes that storage locations returned by the kernel through
+    /// `memop_nr::STORAGE_*` calls are deterministic and in order.
```

Comment:
nit: Actually I realize now that "in order" implies "deterministic", so we could change into "in address space order" or something like that and drop "deterministic".

Reply:
Done